### PR TITLE
Add Ledger app unit tests to the Namada CI

### PR DESCRIPTION
**Pull request by murisi**
## Describe your changes
An attempt to add Ledger app unit tests to the Namada CI. These tests were created to complement the Ledger app E2E tests already in the CI at `test-e2e-with-device-automation`. Whereas the E2E tests comprehensively cover testing hardware wallet signing for a few cases, these unit tests focus on covering hardware wallet printing for hundreds of cases. More specifically, the following changes have been made:
* Made the CI automatically generate random transaction test vectors using the Namada client
  * Doing this has the side benefit of preventing code rot in the test vector generation code
* The CI then runs the Ledger app unit test code on the generated test vectors
  * Doing this helps automatically detect compatibility breaking changes in the transaction serializations
* Fixed regressions in the test vector generation code

See an example run of this new job here: https://github.com/murisi/namada/actions/runs/12143664343/job/33861086521?pr=2

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes
- [ ] If this PR requires changes to the docs or specs, a corresponding PR is opened in the `namada-docs` repo
    - Relevant PR if applies: 
- [ ] If this PR affects services such as `namada-indexer` or `namada-masp-indexer`, a corresponding PR is opened in that repo
    - Relevant PR if applies:

### CODE REVIEW

File: crates/sdk/src/lib.rs [hunk 1]
```diff
@@ -1085,11 +1085,11 @@ pub mod testing {
 
     prop_compose! {
         /// Generate an arbitrary header
-        pub fn arb_header()(
+        pub fn arb_header(cmt_count: impl Into<SizeRange>,)(
             chain_id in arb_chain_id(),
             expiration in option::of(arb_date_time_utc()),
             timestamp in arb_date_time_utc(),
-            batch in arb_tx_commitments(1..10),
+            batch in arb_tx_commitments(cmt_count),
```
Comment (grarco) [inline crates/sdk/src/lib.rs:1092]: Nice one here 👍🏻 , we were previously generating up to 10 headers and the other `arb_transaction` generators were only appending data for a single inner tx which was strange

**Review by grarco — APPROVED** [review]

File: crates/sdk/src/lib.rs [hunk 2]
```diff
@@ -1123,7 +1123,7 @@ pub mod testing {
     prop_compose! {
         /// Generate an arbitrary masp transfer transaction
         pub fn arb_transfer_tx()(
-            mut header in arb_header(),
+            mut header in arb_header(0..=0),
```
Comment (tzemanovic) [inline crates/sdk/src/lib.rs:1126]: nit
```suggestion
            mut header in arb_header(0),
```
it turns into `0..=0`

**Review by tzemanovic — APPROVED** [review]

**Comment by Fraccaman** [comment]
Thanks murisi! I just want to make a couple of changes to the CI workflow before merging

File: crates/sdk/src/lib.rs (hunk 2 above)
Comment (murisi) [reply crates/sdk/src/lib.rs]: Thanks!

**Pull request merged**